Emby.Server.Implementations/HttpServer/Security/AuthorizationContext.cs

```diff
@@ -263,29 +263,60 @@ private AuthorizationInfo GetAuthorization(HttpContext httpReq)
                 return null;
             }
 
-            // Remove uptil the first space
+            // Remove up until the first space
             authorizationHeader = parts[1];
-            parts = authorizationHeader.Split(',');
-
-            var result = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            return GetParts(authorizationHeader);
+        }
 
-            foreach (var item in parts)
+        /// <summary>
+        /// Get the authorization header components.
+        /// </summary>
+        /// <param name="authtorizationHeader">The authorization header.</param>
+        /// <returns>string</returns>
+        public static Dictionary<string, string> GetParts(string authtorizationHeader)
+        {
+            var result = new Dictionary<string, string>();
+            var escapeChars = new[] { '"', ',' };
+            var escaped = false;
+            int start = 0;
+            int i = 0;
+            string key = string.Empty;
+
+            while (i < authtorizationHeader.Length)
```

Comment:
This is should just be a for loop?

Reply:
Should make any difference, since the complexity should be the same. Being more compact I can change it to a for loop.